Products/CMFPlone/browser/favicon.py

```diff
@@ -1,26 +1,44 @@
+from os.path import dirname, join
+
 from Products.CMFPlone.interfaces import ISiteSchema
 from plone.formwidget.namedfile.converter import b64decode_file
-from plone.namedfile.browser import Download
+from plone.memoize import ram
+from plone.namedfile.browser import DisplayFile
 from plone.namedfile.file import NamedImage
 from plone.registry.interfaces import IRegistry
 from zope.component import getUtility
-from plone.memoize import ram
-
 
-class SiteFavicon(Download):
 
+class SiteFavicon(DisplayFile):
     def __init__(self, context, request):
         super().__init__(context, request)
         self.filename = None
         self.data = None
+        # The following attribute disables the use of an allowlist that
+        # would otherwise cause image/vnd.microsoft.icon MIMEtyped files
+        # to be served as downloads.  This allowlist list is sadly not
+        # complete, at the top of the plone.namedfile.browser.py, but
+        # fixing that is beyond the scope of this pull request.
+        self.use_denylist = True
```

Comment:
Minor: This should be a class attribute, there is not point in assigning it every time to the same value